### PR TITLE
Add support for Webtrekk analytics

### PR DESCRIPTION
**Pull request by StephanGuenther**

### CODE REVIEW

**Comment by rudygalfi** [comment]
LGTM.

Some feedback on your documentation at https://supportcenter.webtrekk.com/en/public/amp-analytics.html: Because you've split the explanation into two sections, I think it's possible to read it and understand that you need to use two <amp-analytics> to track pageviews and events. However, that's not the case. It might help to be more explicit that using both just requires merging the two, like this:

```
<amp-analytics type="webtrekk">
  <script type="application/json">
    {
      "vars": {
        "trackDomain": "q3.webtrekk.net",
        "trackId": "111111111111111",
        "contentId": "en.example"
      },
      "triggers": {
        "trackPageview": {
          "on": "visible",
          "request": "pageview"
        },
        "trackClicks": {
          "on": "click",
          "selector": "a",
          "request": "event",
          "vars": {
            "actionName": "click"
          }
        }
      }
    }
  </script>
</amp-analytics>
```

In any case, that's not relevant to this PR.

**Comment by rudygalfi** [comment]
cc @avimehta Once you've taken a look and agreed this is OK, we can merge.

File: extensions/amp-analytics/0.1/vendors.js [hunk 1]
```diff
@@ -297,5 +297,40 @@ export const ANALYTICS_CONFIG = {
...
+  'webtrekk': {
+    'requests': {
+      'trackURL': 'https://${trackDomain}/${trackId}/wt',
+      'parameterPrefix': '?p=431,${contentId},1,' +
+          '${screenWidth}x${screenHeight},${screenColorDepth},' +
+          '${documentReferrer},${timestamp},0,,0&tz=${timezone}' +
+          '&eid=${clientId(amp-wt3-eid)}&la=${browserLanguage}',
+      'parameterSuffix': '&pu=${canonicalUrl}',
+      'pageParameter': '&cp1=${pageParameter1}' +
+          '&cp2=${pageParameter2}&cp3=${pageParameter3}' +
```
Comment (avimehta) [inline extensions/amp-analytics/0.1/vendors.js]: I am assuming that these can be sometimes specified by the publishers. If that is the case,  you can probably get a cleaner config using https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/amp-analytics.md#extra-url-params 

In that case, you just need to specify a mapping from pageParameter to cp,  pageCategory to cg and actionParameter to ck. Then your customers just need to specify the parameters in extraUrlParams section and they will get included in the requests. 

If this works for you, please make the change. If not, just let us know.

Comment (StephanGuenther) [inline extensions/amp-analytics/0.1/vendors.js]: It ist OK for me, in the first step.

**Comment by sebastianbenz** [comment]
@cramforce PTAL

**Comment by cramforce** [comment]
Looks good, please squash commits and rebase with master, so we can merge.

**Comment by avimehta** [comment]
lgtm as well.

**Comment by avimehta** [comment]
@StephanGuenther any updates on this? Please rebase, squash the commits so that we can merge.

thanks

**Comment by cramforce** [comment]
After rebase this will require an entry in `vendor-requests.json`

**Comment by rudygalfi** [comment]
So, just to recap:
* Please rebase
* After rebase this will require an entry in vendor-requests.json
* Please squash all commits

**Comment by avimehta** [comment]
It doesn't look like the commits have been squashed. Are you still working on them? Steps to squash the commits are here: https://github.com/ginatrapani/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

**Comment by StephanGuenther** [comment]
PTAL

**Comment by rudygalfi** [comment]
LGTM

@cramforce Ready to merge

**Pull request merged**